lib/pushmi_pullyu/aip_downloader.rb

```diff
@@ -0,0 +1,147 @@
+require 'pushmi_pullyu/aip_downloader_files_directories'
+require 'pushmi_pullyu/aip_logger'
+require 'pushmi_pullyu/fedora_object_fetcher'
+require 'pushmi_pullyu/solr_fetcher'
+
+# Download all of the metadata/datastreams and associated data
+# related to an object
+
+class PushmiPullyu::AipDownloader
+
+  include PushmiPullyu::AipDownloaderFilesDirectories
+
+  attr_reader :noid, :config, :logger, :aip_logger, :fetcher,
+              :aipcreation_log
+
+  attr_accessor :got_characterization, :got_thumbnail,
+                :got_fedora3foxml, :got_fedora3foxml_metadata
+
+  def initialize(noid, config = nil, application_logger = nil)
+    @noid = noid
+
+    @config = config || PushmiPullyu.options
+    @logger = application_logger || PushmiPullyu.logger
+    @fetcher = PushmiPullyu::FedoraObjectFetcher.new(self.noid, self.config)
+
+    initialize_directory_names
+    initialize_filenames
+
+    # We need to log to application log and to the AIP log
+    @aip_logger = PushmiPullyu::AipLogger.new(@noid, aipcreation_log, @logger)
```

Comment:
A similar comment can be made about adding this stuff to the main application logger: you see a log call and you don't know which files it is writing to! (Which would also be hard to trace.) One of the log files it was writing to may not even exist a minute later. I'm not saying that the current state of this code can't be reworked, I just think you are going down the wrong path by requesting that the main application logger write to two streams (one permanent, the other temporary). At least when you log to "something else", this volatility is better recognized.